engine/player/unique_gear_shadowlands.cpp

```diff
@@ -2831,6 +2831,45 @@ void shard_of_cor( special_effect_t& effect )
   effect.proc_flags2_ = PF2_ALL_HIT | PF2_PERIODIC_DAMAGE;
   new shard_of_cor_cb_t( effect );
 }
+
+/**Shard of Bek
+ * id=355721 driver Rank 1
+ * id=357031 driver Rank 2 (Ominous)
+ * id=357049 driver Rank 3 (Desolate)
+ * id=357058 driver Rank 4 (Foreboding)
+ * id=357069 driver Rank 5 (Portentous)
+ * id=356372 Exsanguinated debuff
+ */
+void shard_of_bek( special_effect_t& effect )
+{
+  struct shard_of_bek_cb_t : public dbc_proc_callback_t
+  {
+    double debuff_value;
+
+    shard_of_bek_cb_t( const special_effect_t& e )
+      : dbc_proc_callback_t( e.player, e ),
+      debuff_value( 0.0001 * e.driver()->effectN( 1 ).average( e.player ) )
+    {
+    }
+
+    void execute( action_t* a, action_state_t* s ) override
+    {
+      dbc_proc_callback_t::execute( a, s );
+
+      auto health_diff = a->player->health_percentage() - s->target->health_percentage();
```

Comment:
I was thinking of adding an RNG roll to this at first but as I said in the description the debuff stays active once it is activated regardless of the current difference so it won't fall if the player gets damaged and the difference becomes smaller and RNG doesn't make sense in that scenario.

Reply:
Yeah, this seems fine. If they change it to fall off, we can add some option to reduce uptime.